File: src/com/oscarparty/servlets/WinnerPickerServlet.java
package com.oscarparty.servlets;

import com.oscarparty.servlets.winners.Categories;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

public class WinnerPickerServlet extends HttpServlet {
    @Override
    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
        //need a list of all categories that don't have a winner
        String[] catsWithoutWinners = Categories.categoriesWithoutWinners();
        req.setAttribute("catsWithoutWinners", catsWithoutWinners);

        req.getServletContext().getRequestDispatcher("/winnerPicker.jsp").forward(req, resp);
    }
}

File: src/com/oscarparty/servlets/WinnerPickerServlet.scala
package com.oscarparty.servlets

import javax.servlet.http.HttpServlet
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse
import com.oscarparty.servlets.data.nominees.OscarNomineesDAO._

import scala.util.parsing.json.{JSONArray, JSONObject}
import com.oscarparty.servlets.data.NextCategory

class WinnerPickerServlet extends HttpServlet {
  protected override def doGet(req: HttpServletRequest, resp: HttpServletResponse) {

    resp.setContentType("text/html; charset=UTF-8")
    resp.setCharacterEncoding("UTF-8")

    val catsToNomsJsonArrays = categoriesWithoutWinners.map { category =>
      category.name -> new JSONArray(category.nominees)
    }.toMap
    val catsToNomsJson = new JSONObject(catsToNomsJsonArrays).toString()

    req.setAttribute("orderedCatsWithoutWinners", categoriesWithoutWinners.map { _.name }.toArray)
    req.setAttribute("catsToNomsMap", catsToNomsJson)
    req.setAttribute("nextCategory", NextCategory.nextCategory)

    req.getServletContext.getRequestDispatcher("/winnerPicker.jsp").forward(req, resp)
  }
}